File: exercism/java/bank-account/src/main/java/BankAccount.java
public class BankAccount {
    private boolean opened = false;
    private int balance = 0;
    public synchronized void open(){
        this.opened = true;
    }
    public synchronized void close(){
        this.opened = false;
    }

    public synchronized int getBalance() throws BankAccountActionInvalidException {
        if(!this.opened){
            throw new BankAccountActionInvalidException("Account closed");
        }
        return balance;
    }

    public synchronized void deposit(int amount) throws BankAccountActionInvalidException {
        if(!this.opened){
            throw new BankAccountActionInvalidException("Account closed");
        }
        if(amount < 0){
            throw new BankAccountActionInvalidException("Cannot deposit or withdraw negative amount");
        }
        this.balance += amount;
    }

    public synchronized void withdraw(int amount) throws BankAccountActionInvalidException {
        if(!this.opened){
            throw new BankAccountActionInvalidException("Account closed");
        }
        if(amount < 0){
            throw new BankAccountActionInvalidException("Cannot deposit or withdraw negative amount");
        }
        if(this.balance <= 0){
            throw new BankAccountActionInvalidException("Cannot withdraw money from an empty account");
        }
        else if(this.balance < amount){
            throw  new BankAccountActionInvalidException("Cannot withdraw more money than is currently in the account");
        }

        this.balance -= amount;
    }
}

File: exercism/clojure/bank-account/src/bank_account.clj
(ns bank-account)

(defn open-account [] ;; <- arglist goes here
  (ref 0)
  )

(defn close-account [bank-account] ;; <- arglist goes here
  (dosync (ref-set bank-account nil))
  )

(defn get-balance [bank-account] ;; <- arglist goes here
  @bank-account
  )

(defn update-balance [bank-account, amount] ;; <- arglist goes here
  (dosync (commute bank-account + amount))
  )
